source/libNsdl/src/sn_grs.c

```diff
@@ -441,71 +579,27 @@ extern int8_t sn_grs_process_coap(struct nsdl_s *nsdl_handle, sn_coap_hdr_s *coa
             } else {
                 /* Static resource handling */
                 switch (coap_packet_ptr->msg_code) {
-                    case (COAP_MSG_CODE_REQUEST_GET):
-                        if (resource_temp_ptr->access & SN_GRS_GET_ALLOWED) {
+                    case COAP_MSG_CODE_REQUEST_GET:
+                        if (resource_temp_ptr->static_resource_parameters->access & SN_GRS_GET_ALLOWED) {
                             status = COAP_MSG_CODE_RESPONSE_CONTENT;
                             static_get_request = true;
                         } else {
                             status = COAP_MSG_CODE_RESPONSE_METHOD_NOT_ALLOWED;
                         }
                         break;
-                    case (COAP_MSG_CODE_REQUEST_POST):
-                        if (resource_temp_ptr->access & SN_GRS_POST_ALLOWED) {
-                            resource_temp_ptr->resourcelen = coap_packet_ptr->payload_len;
-                            handle->sn_grs_free(resource_temp_ptr->resource);
-                            resource_temp_ptr->resource = 0;
-                            if (resource_temp_ptr->resourcelen) {
-                                resource_temp_ptr->resource = handle->sn_grs_alloc(resource_temp_ptr->resourcelen);
-                                if (!resource_temp_ptr->resource) {
-                                    status = COAP_MSG_CODE_RESPONSE_INTERNAL_SERVER_ERROR;
-                                    break;
-                                }
-                                memcpy(resource_temp_ptr->resource, coap_packet_ptr->payload_ptr, resource_temp_ptr->resourcelen);
-                            }
-                            if (coap_packet_ptr->content_format != COAP_CT_NONE) {
-                                if (resource_temp_ptr->resource_parameters_ptr) {
-                                    resource_temp_ptr->resource_parameters_ptr->coap_content_type = coap_packet_ptr->content_format;
-                                }
-                            }
-                            status = COAP_MSG_CODE_RESPONSE_CHANGED;
-                        } else {
```

Comment:
This part is for static resource handling, agreed with Yogesh that only GET should be supported in this case.